src/components/Identity/visitorService/awaitVisitorOptIn.js

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { isObject } from "../../../utils";
+
+export default ({ logger }) => {
+  return new Promise((resolve, reject) => {
+    if (isObject(window.adobe) && isObject(window.adobe.optIn)) {
+      const optInOld = window.adobe.optIn;
+      logger.log(
+        "Delaying request while waiting for legacy opt in to let Visitor retrieve ECID from server."
+      );
+      optInOld.fetchPermissions(() => {
+        if (optInOld.isApproved([optInOld.Categories.ECID])) {
+          logger.log(
+            "Received legacy opt in approval to let Visitor retrieve ECID from server."
+          );
+          resolve();
+        }
+        reject();
```

Comment:
In the case where the user declined consent in Visitor, we're rejecting this promise which, if we follow the rejection up the promise chain, is going to cause the command that initiated the process to be rejected, I believe. Usually when we reject promises like this, we reject it with an error instance and a message explaining what happened. That would mean in this case we would do something like:

```
reject(new Error("Legacy opt-in was declined blah blah...");
```

Once that's in place, test it out manually in the sandbox to see if the command gets rejected:

```
alloy("event", {...}).catch(function(error) {
  // error.message should be the message you provided.
});
```

Then, if you haven't already, write up test cases in TestRail for all these scenarios so that we can write e2e tests for them.